app/javascript/components/pxe-image-type-form/index.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState, useMemo, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import MiqFormRenderer from '@@ddf';
+import createSchema from './pxe-image-type-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+
+const PxeImageForm = ({ recordId }) => {
+  const [{ fields, initialValues, isLoading }, setState] = useState({ fields: [], isLoading: !!recordId });
+  const submitLabel = !!recordId ? __('Save') : __('Add');
+
+  const loadSchema = (appendState = {}) => ({ data: { form_schema: { fields } } }) => {
+    setState((state) => ({
+      ...state,
+      ...appendState,
+      fields,
+    }));
+  };
+
+  useEffect(() => {
+    if (recordId) {
+      API.get(`/api/pxe_image_types/${recordId}`).then((initialValues) => {
+        API.options(`/api/pxe_image_types?ems_id=${initialValues.ems_id}`).then(loadSchema({ initialValues, isLoading: false }));
+      });
+    }
+  }, [recordId]);
+
+  const onSubmit = ({ edit: _edit, ...values }) => {
+    miqSparkleOn();
+
+    const request = recordId ? API.patch(`/api/pxe_image_types`, values) : API.post('/api/pxe_image_types', values);
+    request.then(() => {
+      const message = sprintf(
+        recordId
+          ? __('Modification of System Image Type "%s" has been successfully queued.')
+          : __('System Image Type "%s" was added.'),
+        values.name,
+      );
+      miqRedirectBack(message, undefined, '/pxe/explorer');
+    }).catch(miqSparkleOff);
+  };
+
+  const onCancel = () => {
+    const message = sprintf(
+      recordId
+        ? __('Edit of System Image Type "%s" was cancelled by the user')
+        : __('Add of new System Image Type was cancelled by the user'),
+      initialValues && initialValues.name,
+    );
+    miqRedirectBack(message, 'warning', '/pxe/explorer');
+  };
+
+  return !isLoading && (
+    <MiqFormRenderer
+      schema={createSchema(fields, !!recordId, loadSchema)}
+      initialValues={initialValues}
+      canReset={!!recordId}
+      onSubmit={onSubmit}
+      onReset={() => add_flash(__('All changes have been reset'), 'warn')}
```

Comment:
This is redundant with https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/javascript/forms/data-driven-form.jsx#L79 so you can just delete it.